dat/ai/advertiser.lua

```diff
@@ -0,0 +1,91 @@
+require("dat/ai/tpl/generic.lua")
+require("dat/ai/personality/advertiser.lua")
+
+
+-- Sends a distress signal which causes faction loss
+function sos ()
+   msg = {
+      _("Mayday! We are under attack!"),
+      _("Requesting assistance. We are under attack!"),
+      _("Civilian vessel under attack! Requesting help!"),
+      _("Help! Ship under fire!"),
+      _("Taking hostile fire! Need assistance!"),
+      _("We are under attack, require support!"),
+      _("Mayday! Ship taking damage!"),
+      string.format(_("Mayday! Civilian %s being assaulted!"), string.lower( ai.pilot():ship():class() ))
+   }
+   ai.settarget( ai.target() )
+   ai.distress( msg[ rnd.int(1,#msg) ])
+end
+
+
+mem.shield_run = 100
+mem.armour_run = 100
+mem.defensive  = false
+mem.enemyclose = 500
+mem.distressmsgfunc = sos
+mem.careful   = true
+
+
+function create ()
+
+   -- Credits.
+   ai.setcredits( rnd.int(ai.pilot():ship():price()/500, ai.pilot():ship():price()/200) )
+
+   -- No bribe
+   local bribe_msg = {
+      _("\"Just leave me alone!\""),
+      _("\"What do you want from me!?\""),
+      _("\"Get away from me!\"")
+   }
+   mem.bribe_no = bribe_msg[ rnd.int(1,#bribe_msg) ]
+
+   -- Refuel
+   mem.refuel = rnd.rnd( 1000, 3000 )
+   p = player.pilot()
+   if p:exists() then
+      standing = ai.getstanding( p ) or -1
+      mem.refuel_msg = string.format(_("\"I'll supply your ship with fuel for %d credits.\""),
+            mem.refuel);
+   end
+
+   -- Selects an advertiser message
+   msg = {
+      _("Fly safe, fly Milspec."),
+      _("Reynir's Hot Dogs: enjoy the authentic taste of tradition."),
+      _("Everyone is faster that light, but only Tricon engines are faster than thought!"),
+      _("Dare excellence! Dare Teracom rockets!"),
+      _("Because being unique is hard at times, Nexus invented the Shark fighter."),
+      _("Never take off without your courage. Never take off without your Vendetta."),
+      _("Unicorp: low price and high quality!")
+   }
+
+   -- Faction specific messages
+   if system.cur():presence(faction.get("Empire")) > 1 then
```

Comment:
Seems like a lot of messages are being duplicated. Might be better to try to organize it into a common file.